pkg/operator/ceph/cluster/cleanup.go

```diff
@@ -129,25 +129,28 @@ func (c *ClusterController) cleanUpJobTemplateSpec(cluster *cephv1.CephCluster,
 	return podSpec
 }
 
-func (c *ClusterController) waitForCephDaemonCleanUp(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+func (c *ClusterController) waitForCephDaemonCleanUp(stopCleanupCh chan struct{}, cluster *cephv1.CephCluster, retryInterval time.Duration) error {
 	logger.Infof("waiting for all the ceph daemons to be cleaned up in the cluster %q", cluster.Namespace)
 	for {
-		cephHosts, err := c.getCephHosts(cluster.Namespace)
-		if err != nil {
-			return errors.Wrap(err, "failed to list ceph daemon nodes")
-		}
+		select {
+		case <-time.After(retryInterval):
```

Comment:
Do we really need the loop here? What about letting the reconcile loop run?

Reply:
This method is inside a goroutine, so the reconcile loop is not blocked.